web/src/buddy_list.ts

```diff
@@ -972,7 +974,7 @@ export class BuddyList extends BuddyListConf {
             html,
             new_user_id,
             is_subscribed_user,
-            is_participant_user: this.render_data.all_participant_ids.has(user_id),
+            is_participant_user: this.render_data.all_participant_ids().has(user_id),
```

Comment:
How certain are we that this is never called in a loop? 

It's probably OK in practice, but from an algorithms design standpoint, we shouldn't be computing the full set of participants every time we want to check if a single user is in one.

Probably we can just write this comment:

> Computing the full set of participants just to check if this individual user is a member can in theory do quadratic work, if this function is called in a loop over users. In practice, this is a single-user rerender code path, and the total size of the participants set is unlikely to be thousands of users.

Reply:
(There's several ways to fix this, but I've at least not come up with that one doesn't seem like more of a pain than it's worth)

Reply:
Hmm, `rerender_participants` does call this in a loop though. Maybe `insert_or_move` should take an optional `all_participants_ids` set parameter that the caller can pass to avoid quadratic work.

Reply:
Okay